react/react-training/src/redux/login/actions.js

```diff
@@ -0,0 +1,63 @@
+import UserService from '@services/UserService.js';
+import { MSG_UNKNOWN_ID, ERROR_READING_RESPONSE, ERROR_LOGIN_SERVICE } from '@screens/Login/validation';
+
+export const actionTypes = {
+  LOGIN: 'LOGIN',
+  LOGIN_SUCCESS: 'LOGIN_SUCCESS',
+  LOGIN_FAILURE: 'LOGIN_FAILURE'
+};
+
+export const actionsCreators = {
+  login: (username, password) => async dispatch => {
+    dispatch({ type: actionTypes.LOGIN });
+    const response = await UserService.login(username, password);
+    if (response && response.data) {
+      if (response.ok) {
+        const userIsLogged = true;
+        const userLoginError = null;
+        const userSession = response.data.id || MSG_UNKNOWN_ID;
+        localStorage.setItem('userIsLogged', true.toString());
+        localStorage.setItem('userSession', userSession);
+        dispatch({
+          type: actionTypes.LOGIN_SUCCESS,
+          payload: {
+            userIsLogged,
+            userLoginError,
+            userSession
+          }
+        });
+      } else {
+        const userIsLogged = false;
+        const userLoginError =
+          response.data.error && response.data.error.message && response.data.error.statusCode
+            ? `Error ${response.data.error.statusCode} - ${response.data.error.message}`
+            : ERROR_READING_RESPONSE;
+        const userSession = null;
+        localStorage.removeItem('userIsLogged');
+        localStorage.removeItem('userSession');
+        dispatch({
+          type: actionTypes.LOGIN_FAILURE,
+          payload: {
+            userIsLogged,
+            userLoginError,
+            userSession
+          }
+        });
+      }
+    } else {
+      const userIsLogged = false;
+      const userLoginError = ERROR_LOGIN_SERVICE;
+      const userSession = null;
+      localStorage.removeItem('userIsLogged');
+      localStorage.removeItem('userSession');
+      dispatch({
```

Comment:
You're dispatch the same in all the conditionals the only change is the `type`, i recommend you change this